airbyte-webapp/src/core/jsonSchema/schemaToUiWidget.ts

```diff
@@ -119,12 +121,12 @@ const defaultFields: Array<keyof AirbyteJSONSchema> = [
 
   // airbyte specific fields
   "airbyte_hidden",
-];
+] as const;
 
-const pickDefaultFields = (schema: AirbyteJSONSchema): Partial<AirbyteJSONSchema> => {
-  const partialSchema: Partial<AirbyteJSONSchema> = {
-    ...Object.fromEntries(Object.entries(schema).filter(([k]) => defaultFields.includes(k as keyof AirbyteJSONSchema))),
-  };
+type DefaultFields = Pick<AirbyteJSONSchema, typeof defaultFields[number] | "enum">;
```

Comment:
A couple questions here:
1. What is `typeof defaultFields[number]` doing? It seems like this is just trying to pick all of the fields in the `defaultFields` array from `AirbyteJSONSchema`, but I don't know what that `number` in the brackets is being defined. Is this some functionality specific to `Pick`?
2. Why not just add `"enum"` to the `defaultFields` array? This way you wouldn't need to do ` | "enum"` unless I've missed something